src/katsdpmodels/models.py

```diff
@@ -207,6 +220,43 @@ def from_file(cls: Type[_H], file: io.IOBase, url: str, *,
                 else:
                     return model
 
+    def to_file(self, file: Union[str, pathlib.Path, _FileLike], *,
+                content_type: Optional[str] = None) -> None:
+        if self.version is None:
+            raise ValueError('Version must be set before writing file')
+        if content_type is not None:
+            if content_type != 'application/x-hdf5':
+                raise FileTypeError(f'Expected application/x-hdf5, not {content_type}')
+        else:
+            if isinstance(file, (str, pathlib.Path)):
+                path = pathlib.Path(file)
+            else:
+                path = pathlib.Path(file.name)   # type: ignore
+            if path.suffix not in {'.h5', '.hdf5'}:
+                raise FileTypeError(
+                    f'Expected extension of .h5 or .hdf5, not {path.suffix} '
+                    '(use content_type to override if necessary)'
+                )
+        if isinstance(file, (str, pathlib.Path)):
```

Comment:
Presumably you only strictly need to do this for isinstance(file, pathlib.Path) ?

Reply:
Good point, I'll change it.